docs/development/setup-advanced.md

```diff
@@ -58,51 +58,70 @@ ignoring the parts about `vagrant` (since you're not using it).
+(https://docs.microsoft.com/en-us/windows/wsl/wsl2-about)
 installation method described here.
 
-1. Install WSL 2 by following the instructions provided by Microsoft
-   [here](https://docs.microsoft.com/en-us/windows/wsl/install).
+1. Enable virtualization through your BIOS settings.
+   This sequence depends on your specific hardware and brand, but here
+   are [some basic instructions.]
+   (https://www.thewindowsclub.com/disable-hardware-virtualization-in-windows-10)
 
-1. Install the `Ubuntu 18.04` Linux distribution from the Microsoft
-   Store.
+1. Install WSL 2 by following the instructions provided by Microsoft
+   [here](https://docs.microsoft.com/en-us/windows/wsl/setup/environment).
 
-1. Launch the `Ubuntu 18.04` shell and run the following commands:
+1. Launch the Ubuntu shell as an administrator and run the following command:
 
    ```bash
    sudo apt update && sudo apt upgrade
+   ```
+
+1. Create a new SSH-2 Key for the WSL-2 Virtual Machine and add it to your
+   github account. [Instructions]
+   (https://docs.github.com/en/authentication/connecting-to-github-with-ssh/adding-a-new-ssh-key-to-your-github-account)
+   A SSH key linked to your current computer will not work within the virtual machine.
+
+1. Install the server with the following command:
+
+   ```bash
    sudo apt install rabbitmq-server memcached redis-server postgresql
    ```
 
 1. Open `/etc/rabbitmq/rabbitmq-env.conf` using e.g.:
 
    ```bash
-   sudo vim /etc/rabbitmq/rabbitmq-env.conf
+   sudo nano /etc/rabbitmq/rabbitmq-env.conf
    ```
 
-   Add the following lines at the end of your file and save:
+   Confirm the following lines are at the end of your file,
+   and add them if not present. Then save and exit.
 
    ```ini
    NODE_IP_ADDRESS=127.0.0.1
    NODE_PORT=5672
    ```
 
-1. Make sure you are inside the WSL disk and not in a Windows mounted disk.
+1. Run the command below to make sure you are inside the WSL disk and not in a Windows mounted disk.
    You will run into permission issues if you run `provision` from `zulip`
    in a Windows mounted disk.
 
    ```bash
    cd ~  # or cd /home/USERNAME
    ```
 
-1. [Clone your fork of the Zulip repository][zulip-rtd-git-cloning]
-   and [connecting the Zulip upstream repository][zulip-rtd-git-connect]:
+1. Create your fork of the Zulip repository at [GitHub](github.com/zulip)
+   The main server app is zulip/zulip in the top left-hand box. [Instructions]
+   (https://docs.github.com/en/get-started/quickstart/fork-a-repo)
 
-   ```bash
-   git clone --config pull.rebase git@github.com:YOURUSERNAME/zulip.git ~/zulip
-   cd zulip
-   git remote add -f upstream https://github.com/zulip/zulip.git
-   ```
+1. Connect to the Zulip upstream repository:
+
+```bash
+git clone --config pull.rebase git@github.com:YOURUSERNAME/zulip.git ~/zulip
+cd zulip
+git remote add -f upstream https://github.com/zulip/zulip.git
+```
+
+[Learn more about using Git with Zulip code][zulip-rtd-git-cloning]
```

Comment:
These steps are first cloning your fork, and then connecting to the upstream repo, not only the latter.